### PR TITLE
nalu-wind: updates

### PR DESCRIPTION
**Pull request by jrood-nrel**

### CODE REVIEW

File: var/spack/repos/builtin/packages/nalu-wind/package.py [hunk 1]
```diff
@@ -98,9 +83,7 @@ class NaluWind(CMakePackage, CudaPackage, ROCmPackage):
         )
     for _arch in ROCmPackage.amdgpu_targets:
         depends_on(
-            "trilinos@13.4.0.2022.10.27: "
-            "~shared+exodus+tpetra+zoltan+stk~superlu-dist~superlu"
-            "+hdf5+shards~hypre+gtest+rocm amdgpu_target={0}".format(_arch),
+            "trilinos@13.4.0.2022.10.27: ~shared+rocm amdgpu_target={0}".format(_arch),
```
Comment (psakievich) [inline var/spack/repos/builtin/packages/nalu-wind/package.py]:
```suggestion
            "trilinos@13.4: ~shared+rocm amdgpu_target={0}".format(_arch),
```

**Review by psakievich — COMMENTED** [review]
One of our project specific versions snuck in there.

**Review by psakievich — DISMISSED** [review]

File: var/spack/repos/builtin/packages/nalu-wind/package.py [hunk 2]
```diff
@@ -114,16 +96,30 @@ class NaluWind(CMakePackage, CudaPackage, ROCmPackage):
     # indirect dependency needed to make original concretizer work
     depends_on("netcdf-c+parallel-netcdf")
     depends_on("boost +filesystem +iostreams cxxstd=14", when="+boost")
-    supported_cxxstd = ["17"]
-    variant(
-        "cxxstd", default="17", values=supported_cxxstd, multi=False, description="cxx standard"
+    depends_on("hypre+gpu-aware-mpi", when="+gpu-aware-mpi")
+    depends_on("hypre+umpire", when="+umpire")
+    depends_on("trilinos~shared", when="+trilinos-solvers")
```
Comment (psakievich) [inline var/spack/repos/builtin/packages/nalu-wind/package.py]: Is this still just `darwin` or always?

Comment (jrood-nrel) [reply var/spack/repos/builtin/packages/nalu-wind/package.py]: That's a good question. Maybe @marchdf or @itopcuoglu can answer that.

Comment (itopcuoglu) [reply var/spack/repos/builtin/packages/nalu-wind/package.py]: I have only experienced the related issue on my mac, so I think it is darwin only.

**Review by psakievich — APPROVED** [review]

**Pull request merged**